error.go

```diff
@@ -35,8 +35,19 @@
 //
 // 	err = multierr.Append(reader.Close(), writer.Close())
 //
-// This makes it possible to record resource cleanup failures from deferred
-// blocks with the help of named return values.
+// The underlying list of errors for a returned error object may be retrieved
+// with the Errors function.
+//
+// 	errors := multierr.Errors(err)
+// 	if len(errors) > 0 {
+// 		fmt.Println("The following errors occurred:", errors)
+// 	}
+//
+// Deferred Functions
+//
+// Go makes it possible to modify the return value of a function in a deferral
```

Comment:
I won't change this for now as this sounds like more a general change in the doc comments style and I would rather prefer this to be done from a core maintainer because I would like to get done with this at some time :smile: